flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherResourceCleanupTest.java

```diff
@@ -762,4 +870,40 @@ public JobManagerRunner createJobManagerRunner(
             throw testException;
         }
     }
+
+    private class TestingHistoryServerArchivist implements HistoryServerArchivist {
+
+        private final Executor ioExecutor;
+        private final long sleepMills;
+
+        private boolean archived;
+
+        public TestingHistoryServerArchivist(Executor ioExecutor, long sleepMills) {
+            this.ioExecutor = ioExecutor;
+            this.sleepMills = sleepMills;
+            this.archived = false;
+        }
+
+        public boolean isArchived() {
+            return archived;
+        }
+
+        @Override
+        public CompletableFuture<Acknowledge> archiveExecutionGraph(
+                ExecutionGraphInfo executionGraphInfo) {
+            return CompletableFuture.runAsync(
```

Comment:
You could pass a CompletableFuture as a parameter to this implementation and make it complete in the test code. This way, we wouldn't have to use sleep here...

Reply:
Replaced.